src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,600 @@
+
+    // Initialize the right FootPrint
+    m_pImpl->right = std::make_shared<FootPrint>();
+    m_pImpl->right->setFootName(m_pImpl->names.right);
+
+    // Compute the FootPrints
+    if (!m_pImpl->planner->computeNewSteps(m_pImpl->left,
+                                           m_pImpl->right,
+                                           m_pImpl->horizon.t0,
+                                           m_pImpl->horizon.tf))
+    {
+        m_pImpl->left = nullptr;
+        m_pImpl->right = nullptr;
+        log()->error("{} Failed to compute new steps.", logPrefix);
+        return false;
+    }
+
+    // Create and configure the generator
+    auto generator = UnicycleGenerator();
+    generator.setSwitchOverSwingRatio(m_pImpl->gait.stancePhaseRatio);
+    double lastStepSwitchTime = 0.8;
+    generator.setTerminalHalfSwitchTime(lastStepSwitchTime);
+    generator.setPauseConditions(m_pImpl->duration.max, m_pImpl->duration.nominal);
+
+    // Compute the contact states using the generator
+    if (!generator.generateFromFootPrints(m_pImpl->left,
+                                          m_pImpl->right,
+                                          m_pImpl->horizon.t0,
+                                          m_pImpl->dt.planner))
+    {
+        log()->error("{} Failed to generate from footprints.", logPrefix);
+        return false;
+    }
+
+    // Get the contact states over the horizon
+    std::vector<bool> leftStandingPeriod;
+    std::vector<bool> rightStandingPeriod;
+    generator.getFeetStandingPeriods(leftStandingPeriod, rightStandingPeriod);
+
+    // Lambda to convert the vector with contact states to a vector of PlannedContact.
+    // It just processes timings, we fill the transform later.
+    auto processStandingPeriod
+        = [](const std::vector<bool>& isFootInContactVector,
+             const double dt,
+             const std::string& contactName) -> std::vector<Contacts::PlannedContact> {
+        std::vector<Contacts::PlannedContact> contacts;
+
+        // Loop over all the horizon
+        for (size_t i = 1; i < isFootInContactVector.size(); ++i)
+        {
+            const bool thisState = isFootInContactVector[i];
+            const bool lastState = isFootInContactVector[i - 1];
+            const bool aboutToLand = lastState == 0 && thisState == 1;
+
+            // Create a new contact
+            if (i == 1 || aboutToLand)
+            {
+                auto contact = Contacts::PlannedContact();
+                contact.name = contactName;
+                contact.activationTime = dt * i;
+                contact.deactivationTime = contact.activationTime;
+                contacts.push_back(contact);
+            }
+
+            // Adjust the first contact activation
+            if (i == 1)
+                contacts.back().activationTime -= dt;
+
+            // Upate the contact duration
+            if (thisState)
+                contacts.back().deactivationTime += dt;
```

Comment:
This means that all contacts will be deactivated at the end of the horizon of the planner. Not sure if this is wanted.

Reply:
Do you think that reverting to the previous infinite deactivation time makes more sense? With the old implementation, that was a consequence of the limitations we had, but now we can have this defined. How would it play if we want in the future to implement a merging strategy from previously calculated footsteps?

Reply:
Can you remember me the limitations of using the infinite deactivation time? 

At this point I am not sure what is the best strategy. Using the horizon as deactivation time might cause the robot to jump if the planner is called only once at the beginning :thinking:

Reply:
The limitation was not on the usage of infinite deactivation time, but it was a consequence of what described in https://github.com/dic-iit/bipedal-locomotion-framework/pull/320#discussion_r649992921. For the time being, I can revert to this behavior, and fix it in a future PR when someone will try to make re-planning work with the BLF unicycle implementation.

Reply:
If there are no major limitations, I would go back to the infinite value in case the contact is not deactivated. 

The replanning is just matter of how it is initialized I think. In the end, the output contact list is completely rewritten when calling ``advance``.